project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module

```diff
@@ -23,26 +24,43 @@ function c4m_features_og_group_dashboard_og_permission() {
       'roles'         => array(OG_AUTHENTICATED_ROLE),
       'default role'  => array(OG_ADMINISTRATOR_ROLE),
     ),
+
+    // Recommend flag.
+    'c4m_og_content_recommend' => array(
+      'title'         => t('Recommend group content'),
+      'description'   => t("Allow user to recommend content of the group he's a member of."),
+      'roles'         => array(OG_AUTHENTICATED_ROLE),
+      'default role'  => array(OG_ADMINISTRATOR_ROLE),
+    ),
+    'c4m_og_content_unrecommend' => array(
+      'title'         => t('Unrecommend group content'),
+      'description'   => t("Allow user to unrecommend content of the group he's a member of."),
+      'roles'         => array(OG_AUTHENTICATED_ROLE),
+      'default role'  => array(OG_ADMINISTRATOR_ROLE),
+    ),
   );
 }
 
 /**
  * Implements hook_flag_access().
  */
 function c4m_features_og_group_dashboard_flag_access($flag, $entity_id, $action, $account) {
-  if ($flag->name !== 'c4m_og_content_highlight') {
-    return FALSE;
+  if (!in_array($flag->name, array('c4m_og_content_highlight', 'c4m_og_content_recommend'))) {
+    // Flags we don't care about.
+    return;
```

Comment:
We need to `return` null and not `return FALSE` here because they explicitly check for `FALSE` or `TRUE` in `flag_flag.inc` (line 509). The creator of the flag never think of a flag in the context of `global`. If we'll return `FALSE` instead of `null` the flag will not show on `global` context (that's way kavin didn't see it), when we return `null` the flags (recommend and promote) behave like we expect them to.